oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -52,6 +52,9 @@ public class IdentityPoolCredentials extends ExternalAccountCredentials {
   private static final long serialVersionUID = 2471046175477275881L;
 
   private final IdentityPoolSubjectTokenProvider subjectTokenProvider;
+  // Boolean to tell if a Supplier was provided to build the credential. This is needed to tell
+  // whether the supplier should be passed when creating a new builder from this credential.
+  private final boolean builtWithSupplier;
```

Comment:
We don't need track this -- instead of starting with builder.subjectTokenSupplier != null, move that to the else statement, and then rely on IdentityPoolCredentialSourceType.FILE and IdentityPoolCredentialSourceType.URL.